fastlane/lib/fastlane/helper/xcodes_helper.rb

```diff
@@ -18,9 +18,6 @@ def self.find_xcodes_binary_path
       module Verify
         def self.requirement(req)
           UI.user_error!("Version must be specified") if req.nil? || req.to_s.strip.size == 0
```

Comment:
The `requirement` method was only used in this one place: the `verify_block` in the `xcodes` action, as far as I can tell. I think if we are to remove it, it would probably make sense to remove the whole thing and have it as an inline block in the action's ruby file, which matches other `verify_block` methods. This one was probably pulled out because it was a bit longer and using a rescue block, but generally 1-2 line verification methods are inline.

Reply:
it is still used in two places, therefore I left as it was for reuse

<img width="555" alt="Screenshot 2023-08-05 at 21 23 04" src="https://github.com/fastlane/fastlane/assets/1915802/7115094f-f7f3-4d15-b004-78249236c026">

Reply:
Hmm, I see. So shouldn't `xcversion` continue to use the old validation?

Reply:
I would rather let xcodes or xcversion let decide to throw an error, fastlane should only pre check for empty strings

Reply:
I think I agree with @apps4everyone, the underlying tools will error out if the user enter an invalid version. I think that checking just for string validity (empty/whitespaces) is what makes more sense here 👌